clippy_lints/src/methods/mod.rs

```diff
@@ -2184,6 +2206,58 @@ fn lint_step_by<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, expr: &hir::Expr<'_>, args
     }
 }
 
+fn lint_iter_next<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, expr: &'tcx hir::Expr<'_>, iter_args: &'tcx [hir::Expr<'_>]) {
+    let caller_expr = &iter_args[0];
+
+    let mut parent_expr_opt = get_parent_expr(cx, expr);
```

Comment:
Please leave a comment her why this check is necessary.

Reply:
Okay, will do :+1:

Reply:
I have added the comment.